src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java

```diff
@@ -208,7 +208,9 @@ public void validate() throws PackagerException {
                 Arguments.CLIOptions.PREDEFINED_APP_IMAGE.getId());
         if (appImage != null) {
             Path appImageDir = Path.of(appImage);
-            if (!Files.exists(appImageDir) || appImageDir.toFile().list().length == 0) {
+            if (!Files.exists(appImageDir)
+                    || appImageDir.toFile().list() == null
+                    || appImageDir.toFile().list().length == 0) {
```

Comment:
Are the calls to `list()` idempotent enough here? I.e. can it turn `NULL` on the second call?

Reply:
second call wont be made if first returns null

Reply:
Not what I was asking. Can first call return *not* `NULL`, and then second return `NULL`, failing with NPE? It is customary to use local variable if we are unsure about this.